src/frontend/magnify/src/components/design-system/Formik/ValuesChange/FormikValuesChange.tsx

```diff
@@ -0,0 +1,30 @@
+import * as React from 'react';
+import { FunctionComponent } from 'react';
+import { Form, useFormikContext } from 'formik';
+import { useDebouncedCallback } from 'use-debounce';
+
+interface FormikValuesChangeProps {
+  debounceTime?: number;
+  enableSubmit?: boolean;
+  children?: React.ReactNode;
+}
+
+export const FormikValuesChange: FunctionComponent<FormikValuesChangeProps> = ({ ...props }) => {
+  const formik = useFormikContext();
+  const onValuesChange = useDebouncedCallback(() => {
+    if (props.enableSubmit == null || props.enableSubmit) {
+      formik.handleSubmit();
+    }
+  }, props.debounceTime ?? 800);
+
+  React.useEffect(() => {
+    const values = formik.values;
+    if (values === formik.initialValues) {
```

Comment:
Hmmm... I'm pretty this is not working.

```
const a = { prop: true };
const b = { prop: true };
a === b // -> false
```

`a` & `b` are pointing to two different objects from a memory point of view.

Reply:
I suggest to start by adding a test to demonstrate the problem.